graspologic/partition/leiden.py

```diff
@@ -317,18 +325,23 @@ def leiden(
         weight_default,
         check_directed,
     )
+    if not isinstance(trials, int):
+        raise TypeError("trials must be a positive integer")
+    if trials < 1:
+        raise ValueError("trials must be a positive integer")
```

Comment:
I know we've talked about input checking at various times - I've always thought it could be helpful to use things like https://scikit-learn.org/stable/modules/generated/sklearn.utils.check_scalar.html going forward, just to be more concise and consistent. I don't care for the purposes of this PR but thought I'd bring it up.